File: src/NocWMS-Server/Server.ApiService/Models/Customer.cs
using System;
using System.Collections.Generic;
using KiraNet.Camellia.Infrastructure.DomainModel.Data;
using Server.ApiService.Common;

namespace Server.ApiService.Models
{
    public class Customer : IEntity
    {
        public string Id { get; set; }
        public string CustomerName { get; set; }
        public string Address { get; set; }
        public string Contact { get; set; }
        public Gender Gender { get; set; }
        public string Duty { get; set; }
        public string Telephone { get; set; }
        public string PostCode { get; set; }
        public string Fax { get; set; }
        public string Email { get; set; }
        public string Remarks { get; set; }
        public DateTime CreateTime { get; set; }

        public List<OutboundReceipt> OutboundReceipts { get; set; }
    }
}

File: src/NocWMS-Client/src/app/models/customer.ts
import { Gender } from './gender';

export class Customer {
    constructor(customer?: Partial<Customer>) {
        if (customer) {
            this.id = customer.id;
            this.customerName = customer.customerName;
            this.contact = customer.contact;
            this.duty = customer.duty;
            this.gender = customer.gender;
            this.telephone = customer.telephone;
            this.address = customer.address;
            this.remarks = customer.remarks;
            this.postCode = customer.postCode;
            this.fax = customer.fax;
            this.email = customer.email;
            this.createTime = customer.createTime;
        }
    }

    public id: string;
    public customerName: string;
    public address: string;
    public contact: string;
    public gender: Gender;
    public duty: string;
    public telephone: string;
    public postCode: string;
    public fax: string;
    public email: string;
    public remarks: string;
    public createTime: Date;
}
